scripts/skills/actives/rf_encourage_skill.nut

```diff
@@ -0,0 +1,104 @@
+this.rf_encourage_skill <- this.inherit("scripts/skills/skill", {
+	m = {
+		// Config
+		EncourageBonusFraction = 0.1		// This percentage of Resolve is added as a bonus for the morale check
+	},
+
+	function create()
+	{
+		this.m.ID = "actives.rf_encourage";
+		this.m.Name = "Encourage";
+		this.m.Description = ::Reforged.Mod.Tooltips.parseString("Encourage an ally to raise their current [Morale|Concept.Morale]. A character can only be encouraged once per round. Cannot be used on fleeing or [stunned|Skill+stunned_effect] allies.");
+		this.m.Icon = "ui/perks/perk_28_active.png";	// unused vanilla artwork
+		this.m.IconDisabled = "ui/perks/perk_28_active_sw.png";	// unused vanilla artwork
+		this.m.Overlay = "rf_encourage_skill";
+		this.m.SoundOnUse = [
+			"sounds/combat/inspire_01.wav",		// unused vanilla sound. It is a bit too long and overblown but will do for now
+			"sounds/combat/inspire_02.wav"
+		];
+		this.m.Type = ::Const.SkillType.Active;
+		this.m.Order = ::Const.SkillOrder.UtilityTargeted;
+		this.m.IsSerialized = false;
+		this.m.IsActive = true;
+		this.m.IsTargeted = true;
+		this.m.IsIgnoredAsAOO = true;
+		this.m.ActionPointCost = 3;
+		this.m.FatigueCost = 15;
+		this.m.MinRange = 1;
+		this.m.MaxRange = 3;
+	}
+
+	function getTooltip()
+	{
+		local ret = this.skill.getDefaultUtilityTooltip();
+
+		ret.extend([
+			{
+				id = 10,
+				type = "text",
+				icon = "ui/icons/bravery.png",
+				text = ::Reforged.Mod.Tooltips.parseString("Trigger a positive [Morale Check|Concept.Morale] with a bonus of " + ::MSU.Text.colorizeFraction(this.m.EncourageBonusFraction) + " (" + ::MSU.Text.colorizeValue(this.getEncourageBonus()) + ") of your [Resolve|Concept.Bravery] on the target")
+			},
+			{
+				id = 15,
+				type = "text",
+				icon = "ui/icons/vision.png",
+				text = "Has a range of " + ::MSU.Text.colorGreen(this.getMaxRange()) + " tiles"
+			},
+		]);
+
+		return ret;
+	}
+
+	function onVerifyTarget( _originTile, _targetTile )
+	{
+		if (!this.skill.onVerifyTarget(_originTile, _targetTile)) return false;
+
+		local target = _targetTile.getEntity();
+		if (this.getContainer().getActor().getFaction() != target.getFaction()) return false;
+		return this.__canBeEncouraged(target);
```

Comment:
`return this.getContainer().getActor().getFaction() == target.getFaction() && this.__canbeEncouraged(target);`